tests/mocha/field_variable_test.js

```diff
@@ -8,54 +8,46 @@ suite('Variable Fields', function() {
       }
     };
   }
-  function fieldVariable_createAndInitField(workspace) {
-    var fieldVariable = new Blockly.FieldVariable('name1');
+  function initField(fieldVariable, workspace) {
     var mockBlock = getMockBlock(workspace);
     fieldVariable.setSourceBlock(mockBlock);
     // No view to initialize, but still need to init the model.
     fieldVariable.initModel();
     return fieldVariable;
   }
-
+  function createAndInitFieldConstructor(workspace, variableName) {
+    return initField(new Blockly.FieldVariable(variableName), workspace);
+  }
+  function createAndInitFieldJson(workspace, variableName) {
+    return initField(Blockly.FieldVariable.fromJson(
+        { variable:variableName }), workspace);
+  }
+  function assertValueDefault(variableField) {
+    var actualValue = variableField.getValue();
+    var actualText = variableField.getText();
+    assertNotNull(actualValue);
+    assertNotUndefined(actualValue);
+    assertNotNull(actualText);
+    assertNotUndefined(actualText);
+  }
+  function assertValueName(variableField, expectedText) {
```

Comment:
Ideally you'd want to assert the id and the text are as expected. You may need to replace the method that generates ids for variables with a mock to do this, though. Specificially getOrCreateVariablePackage.

Reply:
I ended up stubbing Blockly.utils.genUid and Blockly.Variables.generateUniqueName instead. It should work the same.